src/vm/database/sqlite.rs

```diff
@@ -259,10 +259,10 @@ impl <'a> ContractDatabase <'a> {
     pub fn set_entry(&mut self, contract_name: &str, map_name: &str, key: Value, value: Value) -> Result<Value> {
         let map_descriptor = self.load_map(contract_name, map_name)?;
```

Comment:
Are these checks performed before the call to `set_entry`?  If not, should they be?  Asking because it confuses me as to why the low-level database code would be responsible for type-checking map inserts and sets.

Reply:
They aren't (apart from the checks done by the type checker).

The reason is that the `map_descriptor` (which is the stored representation of the map definition containing its types) is only available here. That could be refactored to be a public functions so that the caller would be responsible instead.

Reply:
Okay, that's fine for now.  If it continues to bother me, I'll send a separate PR.